catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -131,22 +131,31 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             Map<String, Object> headers) throws SourceUnavailableException, IngestException {
         LOGGER.debug("Creating content item.");
 
-        String key = String.valueOf(ingestedFile.getAbsolutePath()
-                .hashCode());
-        String id = fileIdMap.loadAllKeys()
-                .contains(key) ? String.valueOf(fileIdMap.loadFromPersistence(key)) : null;
-
+        String refKey = (String) headers.get(Constants.STORE_REFERENCE_KEY);
+        String safeKey = null;
+        String id = null;
+        String name = ingestedFile.getName();
+
+        if (refKey != null) {
+            name = FilenameUtils.getName(refKey);
+            // guards against impermissible filesystem characters
+            safeKey = String.valueOf(refKey.hashCode());
+            if (fileIdMap.loadAllKeys()
+                    .contains(safeKey)) {
+                id = String.valueOf(fileIdMap.loadFromPersistence(safeKey));
```

Comment:
Using the reference's hash as the key guards against this. I can add a comment to that effect.